src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -134,7 +134,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Select
 
             Dim expressionInitializer = TryCast(boundInitializer, BoundExpression)
-            Return If(expressionInitializer IsNot Nothing, New BoundEqualsValue(initializer, expressionInitializer), boundInitializer)
+            If expressionInitializer IsNot Nothing Then
+                If Me.MemberSymbol.Kind = SymbolKind.Parameter Then
+                    Return New BoundParameterEqualsValue(initializer, DirectCast(Me.RootBinder.ContainingMember, ParameterSymbol), expressionInitializer)
+                End If
+
+                If Me.MemberSymbol.Kind = SymbolKind.Property Then
+                    Return New BoundPropertyEqualsValue(initializer, DirectCast(Me.MemberSymbol, PropertySymbol), expressionInitializer)
```

Comment:
No. BoundFieldOrPropertyInitializer is not an expression, and one of those serves just as well as a Bound(Field/Property)EqualsValue. See the related discussion below.